frame/support/test/tests/pallet_instance.rs

```diff
@@ -585,196 +589,340 @@ fn pallet_on_genesis() {
-				name: DecodeDifferent::Decoded("MyGetParam".to_string()),
-				ty: DecodeDifferent::Decoded("u32".to_string()),
-				value: DecodeDifferent::Decoded(vec![10, 0, 0, 0]),
-				documentation: DecodeDifferent::Decoded(vec![]),
-			},
-		]),
-		errors: DecodeDifferent::Decoded(vec![
-			ErrorMetadata {
-				name: DecodeDifferent::Decoded("InsufficientProposersBalance".to_string()),
-				documentation: DecodeDifferent::Decoded(vec![
-					" doc comment put into metadata".to_string(),
-				]),
+			],
+		}),
+		calls: Some(PalletCallMetadata {
+			ty: scale_info::meta_type::<pallet::Call<Runtime>>(),
+			calls: vec![
+				FunctionMetadata {
+					name: "foo",
+					arguments: vec![
+						FunctionArgumentMetadata {
+							name: "_foo",
+							ty: scale_info::meta_type::<codec::Compact<u32>>(),
+						}
+					],
+					documentation: vec![
+						" Doc comment put in metadata",
+					],
+				},
+				FunctionMetadata {
+					name: "foo_transactional",
+					arguments: vec![
+						FunctionArgumentMetadata {
+							name: "_foo",
+							ty: scale_info::meta_type::<codec::Compact<u32>>(),
+						}
+					],
+					documentation: vec![
+						" Doc comment put in metadata",
+					],
+				},
+			]
+		}),
+		event: Some(PalletEventMetadata { ty: scale_info::meta_type::<pallet::Event<Runtime>>() }),
+		constants: vec![
+			PalletConstantMetadata {
+				name: "MyGetParam",
+				ty: scale_info::meta_type::<u32>(),
+				value: vec![10, 0, 0, 0],
+				documentation: vec![],
 			},
-		]),
+		],
+		error: Some(PalletErrorMetadata { ty: scale_info::meta_type::<pallet::Error<Runtime>>() }),
 	};
 
-	let mut expected_pallet_instance1_metadata = expected_pallet_metadata.clone();
-	expected_pallet_instance1_metadata.name = DecodeDifferent::Decoded("Instance1Example".to_string());
-	expected_pallet_instance1_metadata.index = 2;
-	match expected_pallet_instance1_metadata.storage {
-		Some(DecodeDifferent::Decoded(ref mut storage_meta)) => {
-			storage_meta.prefix = DecodeDifferent::Decoded("Instance1Example".to_string());
+	let mut example_pallet_instance1_metadata = example_pallet_metadata.clone();
+	example_pallet_instance1_metadata.name = "Instance1Example";
+	example_pallet_instance1_metadata.index = 2;
+	match example_pallet_instance1_metadata.calls {
+		Some(ref mut calls_meta) => {
+			calls_meta.ty = scale_info::meta_type::<pallet::Call<Runtime, pallet::Instance1>>();
+		},
+		_ => unreachable!(),
+	}
+	match example_pallet_instance1_metadata.event {
+		Some(ref mut event_meta) => {
+			event_meta.ty = scale_info::meta_type::<pallet::Event<Runtime, pallet::Instance1>>();
+		},
+		_ => unreachable!(),
+	}
+	match example_pallet_instance1_metadata.error {
+		Some(ref mut error_meta) => {
+			error_meta.ty = scale_info::meta_type::<pallet::Error<Runtime, pallet::Instance1>>();
+		},
+		_ => unreachable!(),
+	}
+	match example_pallet_instance1_metadata.storage {
+		Some(ref mut storage_meta) => {
+			storage_meta.prefix = "Instance1Example";
```

Comment:
These tests were written in a weird way. Not sure why `unwrap` and `as_mut` isn't used here...

Reply:
It is weird, I think it was that way before, but will review and possibly refactor next week.